### PR TITLE
Should not show no index for client rendering bailout

### LINKED ISSUE #58615: Use of useSearchParams is causing the addition of meta robots with noindex in the production build
### Link to the code that reproduces this issue

https://github.com/FrancisGregori/nextjs-noindex-issue

### To Reproduce

1. Use Next.js version `13.5.6-canary.3` or higher.
2. Implement the useSearchParams hook in any page.
3. Build the application for production.
4. Notice that the <meta name="robots" content="noindex"/> tag is automatically added to the generated HTML, even with the correct robots configuration in place.

### Current vs. Expected behavior

**Current Behavior:** 
When using `useSearchParams` in Next.js version 13 or higher, the `<meta name="robots" content="noindex"/>` tag is automatically added to pages in the production build. This occurs even if the robots meta tag is already configured manually, leading to erroneous addition and potential duplication of the robots meta tag.

**Expected Behavior:** 
The `<meta name="robots" content="noindex"/> `tag should only be added in preview versions, not affecting the production build. Manual configurations of the robots meta tags should be respected to avoid duplication and ensure correct indexing of pages by search engines in production.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
Binaries:
  Node: 18.17.0
  npm: 9.6.7
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.0.4-canary.2
  eslint-config-next: 14.0.1
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Metadata (metadata, generateMetadata, next/head), Routing (next/router, next/navigation, next/link)

### Additional context

I have encountered this problem both locally and when deploying on Vercel. Upon further investigation, it seems that this issue started with version `13.5.6-canary.3`. This observation leads me to believe that the erroneous addition of the `<meta name="robots" content="noindex"/>` tag in the production build when using useSearchParams might be specific to this and subsequent versions. Prior versions, including `13.5.6-canary.2` and earlier, do not exhibit this issue.

<sub>[NEXT-1853](https://linear.app/vercel/issue/NEXT-1853/use-of-usesearchparams-is-causing-the-addition-of-meta-robots-with)</sub>

### PR DESCRIPTION
**Pull request by huozhi**
When a static page was built due to bailout to client rendering, like when using `useSearchParams` but without Suspense boundary wrapped on top of it. Since it's expected error on SSR, we don't render extra `noindex` in meta tags to avoid the page being indexed.

Update the test cases of `useSearchparams` hook into dynamic rendering and static-optimized cases: 
* dynanic case will not even have the error id or the meta tag
* static-optimized case in production it won't have this meta tag

Closes NEXT-1853
Fixes #58615

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 6a7bc686018f6ec9d7bec58b3b077292d1f2958f

`pnpm test-dev test/development/app-render-error-log/app-render-error-log.test.ts` 
- app-render-error-log > should log the correct values on app-render error

<details>
<summary>Expand output</summary>

  ● app-render-error-log › should log the correct values on app-render error

    expect(received).toInclude(expected)

    Expected string to include:
      "at fn2 (./app/fn.ts"
    Received:
      " ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.·
       We detected TypeScript in your project and created a tsconfig.json file for you.
     ⚠ `experimental.ppr` has been defaulted to `true` because `__NEXT_EXPERIMENTAL_PPR` was set to `true` during testing.
     ✓ Ready in 3s
     ○ Compiling / ...
     ✓ Compiled / in 2.9s (400 modules)
    found err Error: boom

      at fn2 (webpack-internal:///(rsc)../../../../../../app/fn.ts:6:11)
      at fn1 (webpack-internal:///(rsc)../../../../../../app/fn.ts:9:5)
      at Page (webpack-internal:///(rsc)../../../../../../app/page.tsx:11:45)
      at eW (../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:364883)
      at ../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:378379
      at Array.toJSON (../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:384638)
          at stringify (<anonymous>)
      at ../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:369143
      at e5 (../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:369222)
      at e9 (../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:369723)
      at Timeout._onTimeout (../../../../../../tmp/next-install-4656e13c4ce5783c8f079493d597bd75ec9347502344e5a1bdfb45435fc00528/node_modules/.pnpm/file+..+next-repo-7a60817f6749ce7e830e3a7b00ea02dcb9e016e5a9122f475aeb99ad73651e74+packages+n_dton56r3d7fu264g6lwymwikva/node_modules/next/dist/compiled/next-server/app-page-experimental.runtime.dev.js:35:366058)
      at process.processTimers (../node:internal/timers:514:7) {
        digest: '3184659053'
      }
      "
      at Object.toInclude (development/app-render-error-log/app-render-error-log.test.ts:19:25)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| buildDuration | 10.8s | 10.7s | N/A |
| buildDurationCached | 6s | 5.9s | N/A |
| nodeModulesSize | 200 MB | 200 MB | ⚠️ +6.16 kB |
| nextStartRea..uration (ms) | 422ms | 422ms | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| 170-HASH.js gzip | 26.7 kB | 26.7 kB | N/A |
| 199.HASH.js gzip | 181 B | 185 B | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 242 B | 241 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 98.5 kB | 98.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| _app-HASH.js gzip | 195 B | 194 B | N/A |
| _error-HASH.js gzip | 183 B | 182 B | N/A |
| amp-HASH.js gzip | 501 B | 501 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 255 B | ✓ |
| head-HASH.js gzip | 349 B | 350 B | N/A |
| hooks-HASH.js gzip | 368 B | 369 B | N/A |
| image-HASH.js gzip | 4.27 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.6 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 309 B | N/A |
| script-HASH.js gzip | 384 B | 384 B | ✓ |
| withRouter-HASH.js gzip | 307 B | 306 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| index.html gzip | 530 B | 527 B | N/A |
| link.html gzip | 542 B | 539 B | N/A |
| withRouter.html gzip | 524 B | 523 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.7 kB | 93.7 kB | N/A |
| page.js gzip | 146 kB | 146 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 622 B | N/A |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.4 kB | 37.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.07 kB | 2.07 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js fix/no-index-search-params | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 168 kB | 168 kB | N/A |
| app-page-exp..prod.js gzip | 93.9 kB | 93.9 kB | N/A |
| app-page-tur..prod.js gzip | 94.7 kB | 94.7 kB | N/A |
| app-page-tur..prod.js gzip | 89.2 kB | 89.2 kB | N/A |
| app-page.run...dev.js gzip | 138 kB | 138 kB | N/A |
| app-page.run..prod.js gzip | 88.5 kB | 88.6 kB | N/A |
| app-route-ex...dev.js gzip | 23.9 kB | 23.9 kB | ✓ |
| app-route-ex..prod.js gzip | 16.6 kB | 16.6 kB | ✓ |
| app-route-tu..prod.js gzip | 16.6 kB | 16.6 kB | ✓ |
| app-route-tu..prod.js gzip | 16.2 kB | 16.2 kB | ✓ |
| app-route.ru...dev.js gzip | 23.4 kB | 23.4 kB | ✓ |
| app-route.ru..prod.js gzip | 16.2 kB | 16.2 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| server.runti..prod.js gzip | 49.4 kB | 49.4 kB | ✓ |
| Overall change | 257 kB | 257 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>page.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>170-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>main-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..ntime.dev.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-exp..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page-tur..time.prod.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>app-page.runtime.dev.js</strong></summary>

```diff
failed to diff
```
</details>
<details>
<summary>Diff for <strong>app-page.runtime.prod.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: 6a7bc686018f6ec9d7bec58b3b077292d1f2958f

File: test/e2e/app-dir/hooks/hooks.test.ts [hunk 1]
```diff
@@ -60,6 +60,9 @@ createNextDescribe(
         expect($('#params-second').text()).toBe('other value')
         expect($('#params-third').text()).toBe('')
         expect($('#params-not-real').text()).toBe('N/A')
+
+        // should not have noindex meta tag as it's erroring on purpose, for nextjs internal use only
+        expect($('meta[name=noindex]').length).toBe(0)
```
Comment (ztanner) [inline test/e2e/app-dir/hooks/hooks.test.ts:65]: won't this be true regardless of the changes here since this page doesn't have any errors?

Comment (huozhi) [reply test/e2e/app-dir/hooks/hooks.test.ts]: Nice catch, I separate them into dynamic and static-opt case, so for static-opt case in production it won't have this meta tag. Also a typo on the property 🙈

File: test/e2e/app-dir/hooks/hooks.test.ts [hunk 2]
```diff
@@ -54,12 +54,26 @@ createNextDescribe(
...
         expect($('#params-second').text()).toBe('other value')
         expect($('#params-third').text()).toBe('')
         expect($('#params-not-real').text()).toBe('N/A')
+
+        // dynamic page doesn't have bail out
+        expect($('html#__next_error__').length).toBe(0)
+        expect($('meta[content=noindex]').length).toBe(0)
+      })
+
+      it('should not contain noindex meta tag for static built page', async () => {
```
Comment (ztanner) [inline test/e2e/app-dir/hooks/hooks.test.ts:69]: wdyt about moving this into a separate test? I worry that as we add new unrelated assertions the test apps themselves become bloated (like app-static or parallel-routes-and-interception) so it's harder to understand what's being tested and we may accidentally change behavior of another unrelated test.

**Review by ztanner — APPROVED** [review]

**Pull request merged**